source/Grid/Grid.js

```diff
@@ -740,10 +748,11 @@ export default class Grid extends Component {
     // This causes a series of rapid renders that is slow for long lists.
     // We can avoid that by doing some simple bounds checking to ensure that scrollTop never exceeds the total height.
     const { height, width } = this.props
+    const scrollbarSize = getScrollbarSize()
     const totalRowsHeight = this._getTotalRowsHeight()
     const totalColumnsWidth = this._getTotalColumnsWidth()
-    const scrollLeft = Math.min(totalColumnsWidth - width, event.target.scrollLeft)
-    const scrollTop = Math.min(totalRowsHeight - height, event.target.scrollTop)
+    const scrollLeft = Math.min(totalColumnsWidth - width + scrollbarSize, event.target.scrollLeft)
+    const scrollTop = Math.min(totalRowsHeight - height + scrollbarSize, event.target.scrollTop)
```

Comment:
Thinking about this approach a little more. I think it's safe (based on my testing) but- I want to point out that it doesn't tell us if there is _actually_ a scrollbar or not, just the size of what a scrollbar would be _if_ it were present. (I'm doing a similar check in `FlexTable` but I'm using `offsetWidth` and `clientWidth` directly to determine whether there's a scrollbar.) This approach wouldn't work in `FlexTable` (because it's important whether a scrollbar is _actually_ visible or not).

Okay. Anyway. All of that to say that... since we're not checking whether the scrollbar is actually present or not, only what it's size _would_ be, we might as well just do this calculation only once- in `componentDidMount`. That gets called before the first `onScroll` anyway so it should be safe. We can just cache it as a private property.